breadbox/pyright-ratchet-errors.txt

```diff
@@ -1,21 +1,29 @@
 data_validation.py: error: Argument of type "BinaryIO" cannot be assigned to parameter "filepath_or_buffer" of type "FilePath | ReadCsvBuffer[bytes] | ReadCsvBuffer[str]" in function "read_csv"
 data_validation.py: error: Cannot access attribute "all" for class "bool_t"
+data_validation.py: error: Cannot access attribute "any" for class "ExtensionArray"
 data_validation.py: error: Invalid conditional operand of type "Series | bool_t | Unknown"
 data_validation.py: error: No overloads for "read_csv" match the provided arguments (reportCallIssue)
 dataset.py: error: "value" is not a known attribute of "None" (reportOptionalMemberAccess)
-dataset.py: error: Argument of type "list[dict[str, Series | Unknown | ndarray[Any, Unknown] | Any | NDArray[Unknown]]]" cannot be assigned to parameter "matching_properties" of type "List[Dict[str, str]]" in function "__init__" (reportArgumentType)
-dataset_tasks.py: error: Expression of type "None" is incompatible with declared type "List[Unknown]"
-download_tasks.py: error: Argument of type "ArrayLike" cannot be assigned to parameter "column_labels" of type "List[str] | None" in function "_handle_df_nas" (reportArgumentType)
-env.py: error: Argument of type "str | PathLike[str] | None" cannot be assigned to parameter "fname" of type "StrOrBytesPath | IO[str] | RawConfigParser" in function "fileConfig" (reportArgumentType)
+dataset.py: error: Argument of type "list[dict[str, Series | Unknown | Any | ndarray[Any, Unknown] | NDArray[Unknown]]]" cannot be assigned to parameter "matching_properties" of type "List[Dict[str, str]]" in function "__init__" (reportArgumentType)
+dataset.py: error: Argument of type "type[DatasetFeature]" cannot be assigned to parameter "axis" of type "DatasetFeature | DatasetSample" in function "_get_indexes_by_given_id"
+dataset.py: error: Argument of type "type[DatasetSample]" cannot be assigned to parameter "axis" of type "DatasetFeature | DatasetSample" in function "_get_indexes_by_given_id"
+dataset.py: error: Type "Series | DataFrame* | DataFrame" is not assignable to return type "DataFrame"
+dataset_tasks.py: error: Type "None" is not assignable to declared type "List[Unknown]"
+dataset_uploads_tasks.py: error: Argument of type "Series | DataFrame* | DataFrame" cannot be assigned to parameter "given_ids" of type "Sequence[str]" in function "_get_dimension_labels_and_warnings"
+download_tasks.py: error: Argument of type "ArrayLike" cannot be assigned to parameter "column_labels" of type "List[str] | None" in function "_handle_df_nas"
+download_tasks.py: error: Cannot access attribute "merge" for class "NDArray[Unknown]"
+download_tasks.py: error: Cannot access attribute "merge" for class "ndarray[Any, Unknown]"
+download_tasks.py: error: Cannot access attribute "merge" for class "ndarray[Any, _dtype]"
+env.py: error: Argument of type "str | PathLike[str] | None" cannot be assigned to parameter "fname" of type "StrOrBytesPath | IO[str] | RawConfigParser" in function "fileConfig"
```

Comment:
Some of the differences here are just wording changes in the error descriptions. But the newer version of pyright does also generate a few new errors.